pkg/controller/common/rollout/workloads/deployment_rollout_controller.go

```diff
@@ -109,12 +109,14 @@ func (c *DeploymentRolloutController) VerifySpec(ctx context.Context) (bool, err
 		return false, verifyErr
 	}
 
-	if !c.sourceDeploy.Spec.Paused || getDeployReplicaSize(&c.sourceDeploy) == c.sourceDeploy.Status.Replicas {
+	if !c.sourceDeploy.Spec.Paused ||
+		(getDeployReplicaSize(&c.sourceDeploy) != c.sourceDeploy.Status.Replicas && c.sourceDeploy.Status.Replicas != 0) {
```

Comment:
@ryanzhang-oss Pls help review the changes in deployment rollout/scale controller. The root reason is that, if apply a deployment with `paused: true` & `replicas: 1`, deployment controller will leave its status empty because of `paused: true`. So we should regard a deployment whose `status.replicas` is 0 as not being reconciled or in the middle of scaling. 

That's different from cloneset. Cloneset controller will always create replicas and update status, even if it's `updateStrategy.paused` is true.

Reply:
yes, is there a separate PR?

Reply:
You also need to modify the the corresponding wrong integration test

```
FIt("the deployment don't need to be paused if stable", func() {
			By("Create the source deployment with none")
			var sourceReplica int32 = 6
			sourceDeploy.Spec.Replicas = &sourceReplica
			Expect(k8sClient.Create(ctx, &sourceDeploy)).Should(SatisfyAny(Succeed(), &util.AlreadyExistMatcher{}))
			// test environment doesn't have deployment controller, has to fake it
			c.sourceDeploy.Status.Replicas = sourceReplica // this has to pass batch check
			Expect(k8sClient.Status().Update(ctx, &sourceDeploy)).Should(Succeed())
			targetDeploy.Spec.Replicas = pointer.Int32Ptr(0)
			Expect(k8sClient.Create(ctx, &targetDeploy)).Should(SatisfyAny(Succeed(), &util.AlreadyExistMatcher{}))

			By("verify should not fail b/c of deployment not stable")
			consistent, err := c.VerifySpec(ctx)
			Expect(err).Should(BeNil())
			Expect(consistent).Should(BeTrue())
			Expect(c.rolloutStatus.RolloutTargetSize).Should(BeEquivalentTo(sourceReplica))
			Expect(c.rolloutStatus.NewPodTemplateIdentifier).Should(BeEmpty())
		})

```

Reply:
So if a deploy is paused but `spec.replicas != sts.replicas`, is it okay to proceed with rollout logic?

Reply:
yes, it's ok.

Reply:
Got it. I will fix logic and corresponding test.

Reply:
thx, please make sure that the "initialization" function will mark the deployment as not paused.

Reply:
checked, both deploy rollout & scale controller set `.Spec.Paused = false` correctly.